src/dump.c

```diff
@@ -2014,9 +2014,13 @@ DLLEXPORT jl_value_t *jl_uncompress_ast(jl_lambda_info_t *li, jl_value_t *data)
 
 DLLEXPORT int jl_save_incremental(const char *fname, jl_array_t *worklist)
 {
+    // make temporary file name foo.ji.XXXXXX
+    // (would be better to use mkstemp once this is in libuv -- see libuv/libuv#322)
+    char *tmpfname = mktemp(strcat(strcpy((char *) malloc(strlen(fname)+8), fname), ".XXXXXX"));
```

Comment:
Shouldn't the return value of `mktemp` be checked for `NULL`? If the cache directory is not accessible, `mktemp` will probably fail.